datafusion/sql/src/unparser/plan.rs

```diff
@@ -984,11 +984,18 @@ impl Unparser<'_> {
                 Ok(Some(builder.build()?))
             }
             LogicalPlan::SubqueryAlias(subquery_alias) => {
-                Self::unparse_table_scan_pushdown(
+                let ret = Self::unparse_table_scan_pushdown(
                     &subquery_alias.input,
                     Some(subquery_alias.alias.clone()),
                     already_projected,
-                )
+                );
+                if let Some(alias) = alias {
+                    if let Ok(Some(plan)) = ret {
+                        let plan = LogicalPlanBuilder::new(plan).alias(alias)?.build()?;
+                        return Ok(Some(plan));
+                    }
+                }
```

Comment:
I think you can write this a bit more concisely by unwrapping the error earlier if you wanted:

Something like


```suggestion
                )?;
                if let Some(alias) = alias {
                      let plan = LogicalPlanBuilder::new(plan).alias(alias)?.build()?;
                      return Ok(Some(plan));
                }
                Ok(ret)
```